src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -181,7 +181,9 @@ public void OnStopActivity(Activity activity, object payload)
                 {
                     if (currentStatusCode == ActivityStatusCode.Unset)
                     {
-                        activity.SetStatus(ActivityStatusCode.Error);
+                        // Task cancellation won't trigger the OnException so set the span error information here
+                        activity.SetStatus(ActivityStatusCode.Error, "Operation Canceled");
```

Comment:
@jdt - I left a comment on the [issue](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1812#issuecomment-2135825058). I think we should do this for metric also. Wanted to see if @antonfirsov has any additional thoughts on this.

Reply:
Makes sense to keep this the same, even if it would somewhat change the current implementation. I'll wait for feedback and add it to my PR if we reach an agreement on it.

Reply:
Thanks @antonfirsov. I think Task Cancelled would be a better choice then. @jdt - does that sounds good to you?